lib/iex/lib/iex/introspection.ex

```diff
@@ -659,68 +666,86 @@ defmodule IEx.Introspection do
   Prints the types for the given module and type documentation.
   """
   def t(module) when is_atom(module) do
-    case Typespec.fetch_types(module) do
-      :error ->
-        no_beam(module)
+    case :code.get_doc(module) do
+      {:ok, {:docs_v1, _, :erlang, _, _, _, _} = erl_docs} ->
+        :shell_docs.render_type(module, erl_docs) |> IO.puts()
```

Comment:
Note: I've tried to refactor these  three `t` clauses but each implementation is slightly different  in a way that makes it quite difficult to abstract, so I ended up just nesting the whole implementation in an extra case.